File: frontend-hospital-management/src/services/patientService.js
import api from './api'

const patientService = {
  getAllPatients() {
    return api.get('/patients')
  },

  getPatientById(id) {
    return api.get(`/patients/${id}`)
  },

  createPatient(patient) {
    return api.post('/patients', patient)
  },

  updatePatient(id, patient) {
    return api.put(`/patients/${id}`, patient)
  },

  deletePatient(id) {
    return api.delete(`/patients/${id}`)
  }
}

export default patientService

File: src/main/java/com/enesgunumdogdu/hospital_management/service/PatientService.java
package com.enesgunumdogdu.hospital_management.service;

import com.enesgunumdogdu.hospital_management.domain.dto.PatientDto;

import java.util.List;

public interface PatientService {
    
    List<PatientDto> getAllPatients();
    
    PatientDto getPatientById(Long id);
    
    PatientDto createPatient(PatientDto patientDto);
    
    PatientDto updatePatient(Long id, PatientDto patientDto);
    
    void deletePatient(Long id);
}

